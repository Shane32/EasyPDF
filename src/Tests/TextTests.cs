using System.Drawing;
using System.Text;
using Font = Shane32.EasyPDF.Font;

namespace Tests;

public class TextTests
{
    private readonly PDFWriter _writer = new PDFWriter();

    public TextTests()
    {
        _writer.ScaleMode = ScaleModes.Inches;
        _writer.NewPage(System.Drawing.Printing.PaperKind.Letter, false, 1f, 1f);
        _writer.PrepForTests();
    }

    [Fact]
    public void PositioningTests()
    {
        _writer.Font = new Font(StandardFonts.Helvetica, 10f);
        for (int i = 0; i < 12; i++) {
            _writer.TextAlignment = (TextAlignment)i;
            var pos = new PointF(3f, i * 0.25f);
            var r = 0.0625f;
            _writer.MoveTo(pos).Circle(r);
            _writer.MoveTo(pos).OffsetTo(-r, 0f).LineTo(r + r, 0f);
            _writer.MoveTo(pos).OffsetTo(0f, -r).LineTo(0f, r + r);
            _writer.MoveTo(pos).WriteLine("Alignment: " + _writer.TextAlignment);
        }

        _writer.ToArray().SaveAsPdf().ToASCIIString().RemoveID().ShouldMatchApproved(o => o.NoDiff());
    }

    [Fact]
    public void PositioningTests_StretchedY()
    {
        _writer.Font = new Font(StandardFonts.Helvetica, 5f) { StretchY = 2f };
        for (int i = 0; i < 12; i++) {
            _writer.TextAlignment = (TextAlignment)i;
            var pos = new PointF(3f, i * 0.25f);
            var r = 0.0625f;
            _writer.MoveTo(pos).Circle(r);
            _writer.MoveTo(pos).OffsetTo(-r, 0f).LineTo(r + r, 0f);
            _writer.MoveTo(pos).OffsetTo(0f, -r).LineTo(0f, r + r);
            _writer.MoveTo(pos).WriteLine("Alignment: " + _writer.TextAlignment);
        }

        _writer.ToArray().SaveAsPdf().ToASCIIString().RemoveID().ShouldMatchApproved(o => o.NoDiff());
    }

    [Fact]
    public void PositioningTests_Spacing()
    {
        _writer.Font = new Font(StandardFonts.Helvetica, 10f) { LineSpacing = 1.2f };
        for (int i = 0; i < 12; i++) {
            _writer.TextAlignment = (TextAlignment)i;
            var pos = new PointF(3f, i * 0.25f);
            var r = 0.0625f;
            _writer.MoveTo(pos).Circle(r);
            _writer.MoveTo(pos).OffsetTo(-r, 0f).LineTo(r + r, 0f);
            _writer.MoveTo(pos).OffsetTo(0f, -r).LineTo(0f, r + r);
            _writer.MoveTo(pos).WriteLine("Alignment: " + _writer.TextAlignment);
        }

        _writer.ToArray().SaveAsPdf().ToASCIIString().RemoveID().ShouldMatchApproved(o => o.NoDiff());
    }

    [Fact]
    public void PositioningTests_StretchedY_Spacing()
    {
        _writer.Font = new Font(StandardFonts.Helvetica, 5f) { StretchY = 2f, LineSpacing = 1.2f };
        for (int i = 0; i < 12; i++) {
            _writer.TextAlignment = (TextAlignment)i;
            var pos = new PointF(3f, i * 0.25f);
            var r = 0.0625f;
            _writer.MoveTo(pos).Circle(r);
            _writer.MoveTo(pos).OffsetTo(-r, 0f).LineTo(r + r, 0f);
            _writer.MoveTo(pos).OffsetTo(0f, -r).LineTo(0f, r + r);
            _writer.MoveTo(pos).WriteLine("Alignment: " + _writer.TextAlignment);
        }

        _writer.ToArray().SaveAsPdf().ToASCIIString().RemoveID().ShouldMatchApproved(o => o.NoDiff());
    }

    [Fact]
    public void PositioningTests_StretchedX()
    {
        _writer.Font = new Font(StandardFonts.Helvetica, 10f) { StretchX = 2f };
        for (int i = 0; i < 12; i++) {
            _writer.TextAlignment = (TextAlignment)i;
            var pos = new PointF(3f, i * 0.25f);
            var r = 0.0625f;
            _writer.MoveTo(pos).Circle(r);
            _writer.MoveTo(pos).OffsetTo(-r, 0f).LineTo(r + r, 0f);
            _writer.MoveTo(pos).OffsetTo(0f, -r).LineTo(0f, r + r);
            _writer.MoveTo(pos).WriteLine("Alignment: " + _writer.TextAlignment);
        }

        _writer.ToArray().SaveAsPdf().ToASCIIString().RemoveID().ShouldMatchApproved(o => o.NoDiff());
    }

    [Fact]
    public void PositioningTests_MultiLine()
    {
        _writer.Font = new Font(StandardFonts.Helvetica, 10f);
        for (int i = 0; i < 12; i++) {
            _writer.TextAlignment = (TextAlignment)i;
            var pos = new PointF(3f, i * 0.5f);
            var r = 0.0625f;
            _writer.MoveTo(pos).Circle(r);
            _writer.MoveTo(pos).OffsetTo(-r, 0f).LineTo(r + r, 0f);
            _writer.MoveTo(pos).OffsetTo(0f, -r).LineTo(0f, r + r);
            _writer.MoveTo(pos).WriteLine("Alignment: " + _writer.TextAlignment + "\nLine 2");
        }

        _writer.ToArray().SaveAsPdf().ToASCIIString().RemoveID().ShouldMatchApproved(o => o.NoDiff());
    }

}
