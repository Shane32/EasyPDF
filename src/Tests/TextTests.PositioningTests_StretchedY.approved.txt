%PDF-1.4
%\xE2\xE3\xCF\xD3
2 0 obj
<</Length 890/Filter/FlateDecode>>stream
x\x9C\x8D\x97Mo\xD40\x10\x86\xEF\xF9\x159\xC2a\x83\xC7I\x9C\x98\x1BE\x80\x84\xB8P\xF6\x868\xA0v[Zu[Z*\xF1\xF7q\xE6\xDBe\xDB\xAC\xF8\xD8\xC1\xEB\x19\xBFy\x14\x1E\xCC}s\xB2m\xFA\xD4Ncj\xB7\xE7\xCD\x87m\xF3\xB5\x816\x94\x1F\x1Bh\xA7X~\x86\xF6l\xDF\x84\xF6s\xF9u\xDD\x84\x0E\xDA\xBF\xCD\xF7\x1F\xE5\xFB\xF3&\xE0\xBE\xD3O\\<\\\x1E\xDE\x16c\xE8\xC6R\xED\xB9\x8A\xDD\x90\xDB\x08\xF3\xF21,\x7F\x86\x84\x9FgM\x84\xBE\x1B\x81\x17\xC1\xB6\x02\xF6/\xDF/\xD5\x86Wq\xEFF&ld\x04\xCE\xA5e<\x8E\xB7s\x88\xB3\xE6[#\x03\xF7\x1A\xED\x06Wy\xC8\xBE\x91D\xCB\xF2\xFD\x02\xE8\xCDGh\xC7v{!h"\x1E9u0\xB7\xDB\xBD\x7F\xFC\xB0@|\xF5\xEE\xE6\xEA\xF2v\xBF\xBB}|\xDB~\xD9]<n\xEF~\xBF\xDE^\x13[:\xB1\xF4)\x8D\xE0p\xC4\xC8O\x83\x85\x02\xE1ep\xDB\x81\xA6\x08\x13\x18q\x1Fm_~\xC7)X(\x13Z\xA6\xF3i\xBBd1(.X)\x05\x0Bv\x12\x16\x8C\xB2\xC2%\x96I\xF9\x180\xEFK\xB5{x\xC2\xA6\xBC\x8C\x12\xA1\x9F\xFD\xAB\x12\xE4]\x09\xD5\xCB\x12\x94\x8Dl\x07\x9A"l\xFA\xDE\xB1\xE9\x81\xA7`\xA1lh\x99\x0E\xED\x8DN\x116>XR6\xD8\xC9\xAFLXgS:\x8F\x00s\xF2\xF3\xCF\xEE\xE6\xEAv\xF7\x04\xCD8h\x82194\xE3\xCC\x0F\x85\x85\xA2\xE1ep\xDB\x81\xA6\x08\x9A\x11\x1C\x9A!\x0B\xE0\xEC\xD1\xD02\x1D
\x86\x06\xA7\x08\x1A\x1FlP4\xD8Ih0\xCA
\x9A2=\xF7G\xD1\xB9{|\xBC\xDB?aS\\%\x11\xA6\xC1\xB1\x99\x12?\x15\x16\xCA\x86\x97\xC1m\x07\x9A"lRvl\xD2\xC4S\xB0P6\xB4\x8C\x87\xF2v\xC9bl|\xB0\xA8l\xB0\x93\xD8`\x94\x17\xD8@\x86.\xCD\xED\x94\x8F\xD0\xFD\x8D\xFA_8\xD9l\x97\xBD\x7F\xB3\xE83W\x06\xCE\xA6\xE0\xEC\x1C\x9CM\xC2\xF3\xE4\xE8\xCC#O\xC1B\xE9\xD02\x1E\xCA\xDB%\x8B\xD1\xF1\xC1\xCC\xC3\xD8It\xF2\x8A\x88a\xA6\x07\xE9\x8Fp\x0E\xD19h\x1D\x08\xCE|\xE0\x9D\x0C R\xA6J!\xC9\x17\xE0[\x80g\xA9\x98Ce\xE6\xA0j\x0E\xB5\x9B\x83\x93s\xF0v\x0E\x95\x9E}\xCA\xE0\x04\x1D\xCC\xD0\x14\xEBEbE\x06\xD8~\x14\xAEg\\\x04\xD1t\x08\xD1\x8B\x1Az15U\x06\xAC7Wk\x0B\xF0,\x05\x16\xBD\xAE!\x8A\xAF\xA92`\xD1\x8C--\x9A\xCB\x01\xABR\x9A\xB5\xA9\x9D\x81\xF5+\xDE^^\xB10\xE11\xEB\x82bh\x87\x14\x05\x83i\x12\x06/p\x18\xC4\xE0T\x19\xB2\xC1\x1C\xAE-\xC0\xB3\x14\xD9\x00\xD5\xBF\xFEY\xF1W"\xE7/\xF8x\xA7r\x9A\xA5\xC8\xAA\x94\x83\xBB\x04\x98\xCE)\xD6K\xC8\xA6\xD0M%Kq\xFA\xBA\xB4N\xAF.\x7F\x1D\xBA$%3'$\xEFtH"u\xAA\x0CW2\xADk\x0B\xF0,\xBB+y\xB3\xC3(j\xA7\xCAp\x8D&wi\xD1\\\x0EW\x95\xD2\x04O\xED\x8C+\xAD)\xBEd\xED\x97\x8F\xF1\x08\x8B!\xAE\xC3\x12\x9B\x83\xBB\xBEy\xCF\xC3,\xA2\xA7\xCA\x88\xCD\xA6zm\x01\x9E\xA5\xC4&o{\x98D\xF7T\x19\xB1\xC9\x84/-\x9A\xCB\xDF1}J\x93>\xB53\xB1yM\xFB\xA9\xFC\x1Fc\xC0\xFEcp='\xB1\xEC/\xE2\xD5M<\xE8U<\xD4w\xF1\xE0/\xE3\xDE\xFA\xD9Y?W\xD6\xCFj\xFD\\[?;\xEBgo\xFD\\Y\xDF\xA5\\j\x05\x96\xDD\xBD<\xACY?\xF5\xDD0\xE31\xEB\x12#f\xB5\xC3\xFE\x01\xF9\xD3\xE2\xDE
endstream
endobj
4 0 obj
<</Type/Page/MediaBox[0 0 612 792]/Resources<</ProcSet [/PDF /Text /ImageB /ImageC /ImageI]/Font<</F1 1 0 R>>>>/Contents 2 0 R/Parent 3 0 R>>
endobj
1 0 obj
<</Type/Font/Subtype/Type1/BaseFont/Helvetica/Encoding/WinAnsiEncoding>>
endobj
3 0 obj
<</Type/Pages/Count 1/Kids[4 0 R]/ITXT(3.3.2.0)>>
endobj
5 0 obj
<</Type/Catalog/Pages 3 0 R>>
endobj
6 0 obj
<<>>
endobj
xref
0 7
0000000000 65535 f 
0000001129 00000 n 
0000000015 00000 n 
0000001217 00000 n 
0000000972 00000 n 
0000001282 00000 n 
0000001327 00000 n 
trailer
<</Size 7/Root 5 0 R/Info 6 0 